WordPress/gutenberg
# Remove menubar role and improve complementary area header semantics.

Fixes https://github.com/WordPress/gutenberg/issues/58313

## What?

The Styles panel header contains an ARIA `role="menubar"`. That wraps some buttons. The menubar role is incorrectly used because:
- The button aren't menuitems.
- The menubar doesn't implement arrow navigation

While the intent of the menubar role added in https://github.com/WordPress/gutenberg/pull/51318 was good as it was meant to privide a little more context and an aria-label for the group of buttons, I think this role is incorrectly used for the reasons mentioned above. Also, these buttons are more 'toggle buttons' that toggle other UI parts. They are not command buttons like users would expect in an ARIA rop;e=menu.

Rather, I think the context can be improved by making the visible title in the header a H2 heading.

## Why?

- ARIA roles must be used not only for semantics but also providing the expected keyboard interactionl. The current implementation is not ideal and doesn't provide the expected behavior and semantics.
- All the side panels powered by the `ComplementaryArea` component lack some context. By default, the visible title is only wrapped within a `<strong>` element, whether it's the defautl header or a custom header:
  - Default header: `<strong>{ title }</strong>`
  - Custom header e.g. the one used by the Styles panel: `<strong>{ __( 'Styles' ) }</strong>`
  

## How?

- Removes the menubar role and the associated aria-label.
- Makes the visible text a H2 heading.
- Adjusts the CSS accordingly.

I'll create a follow-up to improve the functionality of the `title` and `smallScreenTitle`, which is used in the 'small' header shown on small screens.

## Testing Instructions




- Go to the Site Editor > Styles 
- Inspect the source and observe the Styles panel header buttons are not wrapped within a role-menubar any longer.
- Observe the `Styles` text  before the buttons is now a H2 heading.
- Observe there's no visual changes.
- Open the Settings panel.
- Observe there's no visual changes.
- Emulate a small screen via dev tools e.g. 768px viewport width.
- Note: the layout suffers from an unrelated bug where some transparent elements overlap the top bar button. As such it's not possible to click these buttons. Use the keyboard to move to them.
- In the Settings panel and the Styles pabel, observe the 'small titles' with text 'Settings' and 'Styles' are now H2 headings.
- Note: as mentioned, I will create a follow-up issue to improve the functionality of these 'small titles' as right now they may just repeat other visible text or be a little misleading.
- Install a plugin that adds its own Plugins sidebar e.g. Yoast SEO.
- Go to the post editor.
- Open the Settings panel. Observ there's no visual changes.
- Open the Plugins panel. Observe the title 'Yoast' is now a H2 heading.
- Emulate a small screen via dev tools e.g. 768px viewport width.
- Open the Settings panel.
- Observe the small title (which in this screen is the post title) is now a H2 heading.
- Open the Plugins panel.
- Observe the small title (which in this screen is the post title) is now a H2 heading.
- Observe there's no visual changes.


### Testing Instructions for Keyboard


## Screenshots or screencast
